acvm-repo/acvm/tests/solver.rs

```diff
@@ -722,3 +724,531 @@ fn memory_operations() {
+        // f(xs, f(ys, zs))
+        let op_ys_zs = bigint_solve_binary_op(bigint_add_op(), modulus.clone(), ys.clone(), zs.clone());
+        let ys_zs: Vec<_> = op_ys_zs.into_iter().map(|x| (x, false)).collect();
+        let op_xs_op_ys_zs = bigint_solve_binary_op(bigint_add_op(), modulus, xs, ys_zs);
+
+        prop_assert_eq!(op_xs_ys_op_zs, op_xs_op_ys_zs)
+    }
+
+    #[test]
+    fn bigint_mul_associative((xs, ys, zs, modulus) in bigint_triple_with_modulus()) {
+        // f(f(xs, ys), zs) ==
+        let op_xs_ys = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs.clone(), ys.clone());
+        let xs_ys: Vec<_> = op_xs_ys.into_iter().map(|x| (x, false)).collect();
+        let op_xs_ys_op_zs = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs_ys, zs.clone());
+
+        // f(xs, f(ys, zs))
+        let op_ys_zs = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), ys.clone(), zs.clone());
+        let ys_zs: Vec<_> = op_ys_zs.into_iter().map(|x| (x, false)).collect();
+        let op_xs_op_ys_zs = bigint_solve_binary_op(bigint_mul_op(), modulus, xs, ys_zs);
+
+        prop_assert_eq!(op_xs_ys_op_zs, op_xs_op_ys_zs)
+    }
+
+    #[test]
+    fn bigint_mul_add_distributive((xs, ys, zs, modulus) in bigint_triple_with_modulus()) {
+        // xs * (ys + zs) ==
+        let add_ys_zs = bigint_solve_binary_op(bigint_add_op(), modulus.clone(), ys.clone(), zs.clone());
+        let add_ys_zs: Vec<_> = add_ys_zs.into_iter().map(|x| (x, false)).collect();
+        let mul_xs_add_ys_zs = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs.clone(), add_ys_zs);
+
+        // xs * ys + xs * zs
+        let mul_xs_ys = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs.clone(), ys);
+        let mul_xs_ys: Vec<_> = mul_xs_ys.into_iter().map(|x| (x, false)).collect();
+        let mul_xs_zs = bigint_solve_binary_op(bigint_mul_op(), modulus.clone(), xs, zs);
+        let mul_xs_zs: Vec<_> = mul_xs_zs.into_iter().map(|x| (x, false)).collect();
+        let add_mul_xs_ys_mul_xs_zs = bigint_solve_binary_op(bigint_add_op(), modulus, mul_xs_ys, mul_xs_zs);
+
+        prop_assert_eq!(mul_xs_add_ys_zs, add_mul_xs_ys_mul_xs_zs)
+    }
+
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_add_zero_l((xs, modulus) in bigint_with_modulus()) {
+        let zero = bigint_zeroed(&xs);
+        let expected_results = drop_use_constant(&xs);
+        let results = bigint_solve_binary_op(bigint_add_op(), modulus, zero, xs);
+
+        prop_assert_eq!(results, expected_results)
+    }
+
+    // TODO(https://github.com/noir-lang/noir/issues/5579): Fails on 49, see bigint_add_zero_l_single_case_49
+    #[test]
+    #[should_panic(expected = "Test failed: assertion failed: `(left == right)`")]
+    fn bigint_add_zero_r((xs, modulus) in bigint_with_modulus()) {
```

Comment:
this test is useless since there is a test for zero 'left' and a test for commutativity